### ISSUE TITLE
getModuleTOMGraph() and getModuleTOM() functions

### ISSUE THREAD

**Issue by SoniaRuiz** [juanbot/CoExpNets#20]
Hi Juan,

I am experiencing an issue when trying to request the TOM data from a specific category and network. What happens is that I am always obtaining the same list of genes within a specific category and network, regardless of the module selected.

I think that the reason might be related to the file path ("out.path" parameter) that I send to the functions getModuleTOMGraph() and getModuleTOM(). I think so because both functions return a message saying that they are reading the TOM file from the installation path, although the parameter 'out.path' is also correctly received:

The following output corresponds to the function 'getModuleTOMGraph()':
> tom = getModuleTOMGraph(tissue="ad",module="plum1",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP", topgenes=2)
Reading from /home_2/gsit/RLibrary/CoExpROSMAP//extdata/netad.8.it.50.rds
[1] "/home_2/gsit/RLibrary/CoExpToms/"

On the other hand, the following output corresponds to the function 'getModuleTOM()':
> tom = getModuleTOM(tissue="ad",module="plum1",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP")
Reading from /home_2/gsit/RLibrary/CoExpROSMAP//extdata/netad.8.it.50.rds
[1] "/home_2/gsit/RLibrary/CoExpToms/"

The path  "/home_2/gsit/RLibrary/CoExpToms/" contains one TOM file per each category, network and module:

![image](https://user-images.githubusercontent.com/43370296/75031621-93d50a80-549e-11ea-85c7-97401f903a21.png)

Let me know if you need any further information.

Sonia

**Comment by SoniaRuiz** [comment]
Hi Juan,

Not sure if you have had time to have a look into this issue. In case you need any further details about this issue, please let me know.

Sonia

**Comment by juanbot** [comment]
Sonia, working on this. I am not sure what is the problem. Do you mean that if you ask for the plum1 and then you ask for the blue module, it always return the same?
Thanks

**Comment by SoniaRuiz** [comment]
The problem is that the function 'getModuleTOMGraph()' seems to be ignoring the parameter 'out.path'. I think so because the function 'getModuleTOMGraph()' always returns the TOM data from the installation path, instead of returning it from the path that I have indicated through the parameter 'out.path'.

The installation path only contains one TOM file per each network (i.e. "netad.8.it.50.rds"), which does not contain specific information about the module. In this sense, the plot graph always receives that type of file (i.e. "netad.8.it.50.rds") regardless of the module selected. And that is the reason why I think that the plot always shows the same information independently of the module selected.

**Comment by juanbot** [comment]
Hi Sonia, it is not exactly like that. When the function says "reading from ..." it means that it is reading the whole network and that is correct because the method has to. I would like to see a new code example in which you call 

tom = getModuleTOMGraph(tissue="ad",module="plum1",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP", topgenes=2)

print(dim(tom))

tom = getModuleTOMGraph(tissue="ad",module="yellow",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP", topgenes=2)

print(dim(tom))


If the method works, the two dimensions should be different. Could you please try this?
getModuleTOMGraph

**Comment by SoniaRuiz** [comment]
Sure!

After loading the packages:

![image](https://user-images.githubusercontent.com/43370296/78244387-63d94800-74dd-11ea-96e8-b444fe0b4402.png)


This is the output that I have obtained:


![image](https://user-images.githubusercontent.com/43370296/78244271-2bd20500-74dd-11ea-952e-df8afff0c209.png)

**Comment by juanbot** [comment]
Sorry, I would ask you one more thing.
Please run this code now.
tom = getModuleTOMGraph(tissue="ad",module="plum1",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP", topgenes=2)

str(tom)
print(tom)
tom = getModuleTOMGraph(tissue="ad",module="yellow",out.path="/home_2/gsit/RLibrary/CoExpToms/",which.one="CoExpROSMAP", topgenes=2)

str(tom)
print(tom)
**********

Just want to check really whether the matrices are identical, but something does not seem right...

**Comment by SoniaRuiz** [comment]
Hi Juan!

I found the error, it was on my side!

Sorry, you were absolutely right, as the function getModuleTOMGraph() was working fine:

![image](https://user-images.githubusercontent.com/43370296/78247457-d0a31100-74e2-11ea-93d0-6df6fc58384b.png)

The error was a really silly one, I was hardcoding the module and always returning the 'black' one, regardless of the network and module selected. Not sure why I didn't realise about it before.

I have updated the test environment and now, the plot function works as expected:

If we select 'notad' and 'cyan', the top 2 genes are 'COMMD7' and 'MRPS24':
![image](https://user-images.githubusercontent.com/43370296/78247761-52933a00-74e3-11ea-92da-9737ef6cede4.png)

On the contrary, if we select 'notad' and 'plum1', the top 2 genes are 'WDR47' and 'DNAJC6':
![image](https://user-images.githubusercontent.com/43370296/78248010-c3d2ed00-74e3-11ea-8e74-32e872039832.png)


Results that match with the ones returned by the function 'getModuleTOMGraph()':
![image](https://user-images.githubusercontent.com/43370296/78248235-288e4780-74e4-11ea-9408-3f23bd601c39.png)
